code/modules/mining/lavaland/megafauna_loot.dm

```diff
@@ -0,0 +1,1011 @@
+	projectile.preparePixelProjectile(attacked_atom, src)
+	projectile.firer = src
+	projectile.fire(null, attacked_atom)
+	visible_message(span_danger("[src] fires at [attacked_atom]!"), span_notice("You fire at [attacked_atom]!"))
+	playsound(src, 'sound/magic/fireball.ogg', 50, TRUE)
+
+/obj/item/soulscythe/proc/slash_target(atom/attacked_atom)
+	if(isliving(attacked_atom) && use_blood(10))
+		var/mob/living/attacked_mob = attacked_atom
+		if(attacked_mob.stat != DEAD)
+			give_blood(15)
+		attacked_mob.apply_damage(damage = force * (ishostile(attacked_mob) ? 2 : 1), sharpness = SHARP_EDGED, bare_wound_bonus = 5)
+		to_chat(attacked_mob, span_userdanger("You're slashed by [src]!"))
+	else if((ismachinery(attacked_atom) || isstructure(attacked_atom)) && use_blood(5))
+		var/obj/attacked_obj = attacked_atom
+		attacked_obj.take_damage(force, BRUTE, MELEE, FALSE)
+	else
+		return
+	COOLDOWN_START(src, attack_cooldown, 1 SECONDS)
+	animate(src)
+	SpinAnimation(5)
+	addtimer(CALLBACK(src, .proc/reset_spin), 1 SECONDS)
+	visible_message(span_danger("[src] slashes [attacked_atom]!"), span_notice("You slash [attacked_atom]!"))
+	playsound(src, 'sound/weapons/bladeslice.ogg', 50, TRUE)
+	do_attack_animation(attacked_atom, ATTACK_EFFECT_SLASH)
+
+/obj/item/soulscythe/proc/charge_target(atom/attacked_atom)
+	if(charging || !use_blood(30))
+		return
+	COOLDOWN_START(src, attack_cooldown, 5 SECONDS)
+	animate(src)
+	charging = TRUE
+	visible_message(span_danger("[src] starts charging..."))
+	balloon_alert(soul, "you start charging...")
+	if(!do_after(soul, 2 SECONDS, target = src, timed_action_flags = IGNORE_TARGET_LOC_CHANGE))
+		balloon_alert(soul, "interrupted!")
+		return
+	visible_message(span_danger("[src] charges at [attacked_atom]!"), span_notice("You charge at [attacked_atom]!"))
+	new /obj/effect/temp_visual/mook_dust(get_turf(src))
+	playsound(src, 'sound/weapons/thudswoosh.ogg', 50, TRUE)
+	SpinAnimation(1)
+	throwforce *= 2
+	throw_at(attacked_atom, 10, 3, soul, FALSE)
+
+/obj/item/soulscythe/proc/reset_spin()
+	animate(src)
+	SpinAnimation(15)
+
+/mob/living/simple_animal/soulscythe
+	name = "mysterious spirit"
+	maxHealth = 200
+	health = 200
+	gender = NEUTER
+	mob_biotypes = MOB_SPIRIT
+	faction = list()
+	/// Blood level, used for movement and abilities in a soulscythe
+	var/blood_level = MAX_BLOOD_LEVEL
+
+/mob/living/simple_animal/soulscythe/get_status_tab_items()
+	. = ..()
+	. += "Blood: [blood_level]/[MAX_BLOOD_LEVEL]"
+
+/mob/living/simple_animal/soulscythe/Life(delta_time, times_fired)
+	. = ..()
+	if(!stat)
+		blood_level = min(MAX_BLOOD_LEVEL, blood_level + round(1 * delta_time))
+
+/obj/projectile/soulscythe
+	name = "soulslash"
+	icon_state = "soulslash"
+	flag = MELEE //jokair
+	damage = 15
+	light_range = 1
+	light_power = 1
+	light_color = LIGHT_COLOR_BLOOD_MAGIC
+
+/obj/projectile/soulscythe/on_hit(atom/target, blocked = FALSE)
+	if(ishostile(target))
+		damage *= 2
```

Comment:
making this review so athath can put the eyes react on it

Reply:
eyes